### ISSUE TITLE
Missing Google Action

### ISSUE THREAD

**Issue by Incendium87** [johnnyruz/OctoPrint-OctoVox#21]
The google action seems to be missing or incomplete. I find it when searching on my phone, but clicking into it shows its blank. When searching on a computer, its found as a listing in the search, but clicking in I get a page saying it can't be found.

Is anyone else seeing this? Is this a google server issue, or something specific to this action?

**Comment by johnnyruz** [comment]
Very interesting. I'm able to see a page populated on my phone and browser so maybe a Google Issue.

What happens if you hit this URL directly: https://assistant.google.com/services/a/uid/0000005054805313

**Comment by Incendium87** [comment]
I get the same thing as before. I've attached screen shots of what I get on my phone and computer. I've tried it in three different browsers on my computer with the same result. Also, I'm now at work so I'm using an entirely different computer and internet connection than I was last night.
![OctoVox Action not found - 2019-10-17_11-39-26](https://user-images.githubusercontent.com/56668434/67029523-16a21f00-f0d3-11e9-81d6-5df2f670bbd0.png)
![IMG_0486](https://user-images.githubusercontent.com/56668434/67029567-2faad000-f0d3-11e9-9962-072a7250aaf7.PNG)

**Comment by johnnyruz** [comment]
If I go incognito in my browser I get the same thing you do. When logged in I see the page correctly. I will look into this to see if there's some setting or anything that I would have checked to not make it public.

**Comment by Incendium87** [comment]
When you mentioned that it shows up for you, I wondered if it might be a public or publishing issue, but the title of the action does show up if I start typing it into the search field. So its partially public? That seems odd to me, but I've had no experience with google actions before.
![OctoVox Action found in search  - 2019-10-17_13-01-16](https://user-images.githubusercontent.com/56668434/67035240-e5c7e700-f0de-11e9-8743-4ddc20daa7c9.png)

**Comment by johnnyruz** [comment]
Yeah there's some funky stuff going on that I'm going to look into tonight. One other Google problem that I'll raise with them is that on my phone/computer the Assistant App hears "My 3D Print Server" and interprets it to "My Three D Print Server" and launches the app correctly. On my Google Home Hub, it interprets it to "My 3D Print Server" and can't find the Action......one team isn't talking to another there it seems!

Will let you know if there's any update from my side.

**Comment by Incendium87** [comment]
That certainly is odd. Hopefully that is easy enough to fix. Not being able to talk to it from a Hub would be pretty limiting.

**Comment by johnnyruz** [comment]
I just logged into my account to try to troubleshoot, and received this message:

![image](https://user-images.githubusercontent.com/7636977/67108293-30f3ff80-f19c-11e9-88b3-a1b1f1658f5c.png)

Still working on what actions I can take to resolve this, but wanted to keep this thread updated with any findings.

**Comment by johnnyruz** [comment]
It looks like this may have been resolved as I'm now able to hit the page in Incognito mode, so please take a look and confirm.

I'm still working with Google on the "3D" vs "Three D" issue since my Home Hub continues to not want to find the Action.

**Comment by Incendium87** [comment]
Yes, I can now use the action. As of last night my home mini didn't understand the trigger properly, but the assistant on my phone did.

This may not be ideal, but would it be simple enough just to rename the action to something easier to understand? Something like 'my printer' or 'my print server'

**Comment by johnnyruz** [comment]
Yes that would be easier, but apparently that would violate policies about having an action name that is too generic. I sent in a support request and got some feedback, but the suggestion they gave (creating an alternate pronunciation of "My 3D Print Server") also failed with the error of "Numbers are not allowed". So trying to get an accurate answer. I've also thought about changing the invocation for both Google and Amazon to "OctoVox 3D" to make things shorter, but will run into the same issues with numeric in the invocation.

**Comment by vannaka** [comment]
I'm also having the same issue with my Home Mini and my phone. It interprets what I'm saying as "3D" and not "Three D" thus this plugin is functionally useless. I relize it's not the devs fault here but googles. What a shame.

**Comment by vannaka** [comment]
I found [this](https://developers.google.com/assistant/discovery/explicit). Looks like we have to use an 'explicit invocation' as a workaround. Ex. "OK Google, Talk to my three(3) d print server" which brings up a sub-menu/prompt where you can then ask for the status or what is printing. This isn't perfect as I have to go through an extra step to get the information I want but it's better than nothing.

**Comment by vannaka** [comment]
Not to spam any more but It randomly started to work. If I say "OK Google, **Ask** My 3d print server what the status of myprinter is." it works. Not what sure what I changed/said differently to make it work. I hope anything I've said here can help someone else. Good luck.

**Comment by acampitos** [comment]
Hi,

I'm facing the same issue, Google Action is not available :(
I'm new to octoprint but I wanted to test your action because seems to be really useful.

Did you consider to rename the action to 'Octovox printer' or something like that to avoid that Google invocation issue?

I'm trying to get the action in Spain region.

Thanks in advance

**Comment by justinmbrock** [comment]
I can find the Three D Print Server with Google Assistant, but can't link my account. It briefly opens a browser window that closes before it loads anything to see. Here are a couple screenshots. Would Google's Advanced Protection Program interfere with this plugin?

![Screenshot_20200330-142852](https://user-images.githubusercontent.com/6379307/77948413-5d579000-7293-11ea-81b7-7f5707141cec.png)
![Screenshot_20200330-142928](https://user-images.githubusercontent.com/6379307/77948457-6f393300-7293-11ea-9107-f7c916d10a1b.png)

**Comment by justinmbrock** [comment]
Whatever the issue was seems to have been fixed for me. I was able to link my account today.

**Comment by sjscicluna83** [comment]
I can link my account by web but can't see it on my IOS device.

![IMG_1497510DD020-1](https://user-images.githubusercontent.com/58419766/84127020-07b51600-aa82-11ea-86d1-c7fd5c01499b.jpeg)

![IMG_1A44CB1FDCFF-1](https://user-images.githubusercontent.com/58419766/84127256-6084ae80-aa82-11ea-8abb-8d5ac9fd76e6.jpeg)

**Comment by ItsVince1234** [comment]
The service isnt showing up at all, whats going on?

**Comment by duhredstonegod** [comment]
yeah the app is not showing up on google at all for me

that direct link pulled up a null

**Comment by artifishvr** [comment]
Dang, hopefully this google mess will get figured out soon

**Comment by howels** [comment]
Still getting a null, any idea if the Google action can be fixed?
